src/Voron/Impl/WeakSliceSmallSet.cs

```diff
@@ -0,0 +1,339 @@
+﻿using System;
+using System.Buffers;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using Sparrow;
+using Sparrow.Server;
+
+namespace Voron.Impl
+{
+    // The concept of the weak small set is about an optimized set that can lose elements when it gets filled.
+    // In a sense it behaves like a forgetful LRU cache, but will not track the read accesses. This version of the SmallSet
+    // is specially designed to deal with sequence values stored in byte pointers. The main difference with a
+    // dictionary is that the small set will calculate the hash ONLY if there are strong candidates that match already. 
+    public sealed class WeakSliceSmallSet<TValue> : IDisposable
```

Comment:
As I can see, no one is actually using this class, can we kill it?

Reply:
I build the two variants because I will be using this to speed up some Voron operations when we can do that.